website/docs/r/security_group.html.markdown

```diff
@@ -25,30 +25,37 @@ Provides a security group resource.
 ```terraform
 resource "aws_security_group" "allow_tls" {
   name        = "allow_tls"
-  description = "Allow TLS inbound traffic"
+  description = "Allow TLS inbound traffic and all outbound traffic"
   vpc_id      = aws_vpc.main.id
 
-  ingress {
-    description      = "TLS from VPC"
-    from_port        = 443
-    to_port          = 443
-    protocol         = "tcp"
-    cidr_blocks      = [aws_vpc.main.cidr_block]
-    ipv6_cidr_blocks = [aws_vpc.main.ipv6_cidr_block]
-  }
-
-  egress {
-    from_port        = 0
-    to_port          = 0
-    protocol         = "-1"
-    cidr_blocks      = ["0.0.0.0/0"]
-    ipv6_cidr_blocks = ["::/0"]
-  }
-
   tags = {
     Name = "allow_tls"
   }
 }
+
+resource "aws_vpc_security_group_ingress_rule" "allow_tls" {
+  security_group_id = aws_security_group.allow_tls.id
+
+  cidr_ipv4   = [aws_vpc.main.cidr_block]
+  cidr_ipv6   = [aws_vpc.main.ipv6_cidr_block]
+  from_port   = 443
+  ip_protocol = "tcp"
+  to_port     = 443
+}
```

Comment:
Similar to `aws_vpc_security_group_egress_rule`, only one of `cidr_ipv4` or `cidr_ipv6` can be used at the same time, else you receive the following error on apply:

```shell
╷
│ Error: Invalid Attribute Combination
│
│   with aws_vpc_security_group_ingress_rule.allow_tls,
│   on main.tf line 16, in resource "aws_vpc_security_group_ingress_rule" "allow_tls":
│   16:   cidr_ipv4         = aws_vpc.main.cidr_block
│
│ Exactly one of these attributes must be configured: [cidr_ipv4,cidr_ipv6,prefix_list_id,referenced_security_group_id]
╵
```

The value for `cidr_ipv4` and `cidr_ipv6` is required to be a string rather than a list, so also corrected that in this suggestion.

```suggestion
resource "aws_vpc_security_group_ingress_rule" "allow_tls_ipv4" {
  security_group_id = aws_security_group.allow_tls.id
  cidr_ipv4         = aws_vpc.main.cidr_block
  from_port         = 443
  ip_protocol       = "tcp"
  to_port           = 443
}

resource "aws_vpc_security_group_ingress_rule" "allow_tls_ipv6" {
  security_group_id = aws_security_group.allow_tls.id
  cidr_ipv6         = aws_vpc.main.ipv6_cidr_block
  from_port         = 443
  ip_protocol       = "tcp"
  to_port           = 443
}
```